rules/etcd.go

```diff
@@ -30,6 +30,48 @@ func (edv3ra *etcdV3ReadAPI) get(key string) (*string, error) {
 	return &val, nil
 }
 
+func (edv3ra *etcdV3ReadAPI) getCachedAPI(keys []string) (readAPI, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	uniqueKeys := make(map[string]bool)
+	// Get rid of duplicates
+	for _, key := range keys {
+		uniqueKeys[key] = true
+	}
+
+	ops := make([]clientv3.Op, len(uniqueKeys))
+	idx := 0
+	for key := range uniqueKeys {
+		ops[idx] = clientv3.OpGet(key)
+		idx++
+	}
+	// An etcd transaction consists of four parts:
+	// 1. A set of "if" conditions.  The empty set evaluates to true, as in this case.
+	// 2. A set of "then" operations which are performed if the "if" condition is true.
+	//    In this case these operations are always performed.
+	// 3. A set of "else" operations which are performed if the "if" condition is not true.
+	//    In this case these operations are never performed.
+	// 4. A commit, which finalizes the transaction and commits it. In this case everything is a read,
+	//    so "commiting" doesn't really do anything to change the state of etcd.
+	txnResp, err := edv3ra.kV.Txn(ctx).If().Then(ops...).Else().Commit()
```

Comment:
This intentionally does not check for batch size.  If the batch is larger than 128, we want it to fail. There should be no such rules.